CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+* [Making a change](#making-a-change)
+* [Structuring contributions](#style-guide)
+* [Recognizing contributors](#recognizing-contributions)
+
+## Joining the conversation
+
+`jupyter-book` is a young project maintained by a growing group of enthusiastic developers&mdash; and we're excited to have you join!
+Most of our discussions will take place on open [issues][link_issues].
+
+## Contributing through GitHub
+
+[git][link_git] is a really useful tool for version control.
+[GitHub][link_github] sits on top of git and supports collaborative and distributed working.
+
+You'll use [Markdown][markdown] to chat in issues and pull requests on GitHub.
+You can think of Markdown as a few little symbols around your text that will allow GitHub
+to render the text with formatting.
+For example you could write words as bold (`**bold**`), or in italics (`*italics*`),
+or as a [link][rick_roll] (`[link](https://https://youtu.be/dQw4w9WgXcQ)`) to another webpage.
+
+GitHub has a helpful page on
+[getting started with writing and formatting Markdown on GitHub][writing_formatting_github].
+
+
+## Understanding issues, milestones and project boards
+
+Every project on GitHub uses [issues][link_issues] slightly different.
+
+The following outlines how the `jupyter-book` developers think about these different tools.
+
+**Issues** are individual pieces of work that need to be completed to move the project forwards.
+A general guideline: if you find yourself tempted to write a great big issue that
+is difficult to describe as one unit of work, please consider splitting it into two or more issues.
+
+Issues are assigned [labels](#issue-labels) which explain how they relate to the overall project's
+goals and immediate next steps.
+
+
+### Issue labels
+
+The current list of labels are [here][link_labels] and include:
+
+* [![Help Wanted](https://img.shields.io/badge/-help%20wanted-159818.svg)][link_helpwanted] *These issues contain a task that a member of the team has determined we need additional help with.*
+
+    If you feel that you can contribute to one of these issues, we especially encourage you to do so!
+
+* [![Bugs](https://img.shields.io/badge/-bugs-fc2929.svg)][link_bugs] *These issues point to problems in the project.*
+
+    If you find new a bug, please give as much detail as possible in your issue, including steps to recreate the error.
+    If you experience the same bug as one already listed, please add any additional information that you have as a comment.
+
+* [![Enhancement](https://img.shields.io/badge/-enhancement-84b6eb.svg)][link_enhancement] *These issues are asking for enhancements to be added to the project.*
+
+    Please try to make sure that your enhancement is distinct from any others that have already been requested or implemented.
+    If you find one that's similar but there are subtle differences please reference the other request in your issue.
```

Comment:
What are your thoughts on "good first issue" @emdupre ? I like the idea of having a relatively small number of total issue tags, so don't wanna suggesting adding a new one if it'll just add noise!

Reply:
I think that's a great tag ! It's nice to demarcate that some issues are looking for new contributors of any experience level vs ones where you specifically think a newcomer could easily hop on. Will add it here.